flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/streamrecord/StreamElementSerializerTest.java

```diff
@@ -89,6 +90,9 @@ public void testSerialization() throws Exception {
 
 		Watermark negativeWatermark = new Watermark(-4647654567676555876L);
 		assertEquals(negativeWatermark, serializeAndDeserialize(negativeWatermark, serializer));
+
+		LatencyMarker latencyMarker = new LatencyMarker(System.currentTimeMillis(), new OperatorID(-1, -1), 1);
+		assertEquals(latencyMarker, serializeAndDeserialize(latencyMarker, serializer));
```

Comment:
hashCode and equals changes are to support this line here. Without it, even though the objects have the same values, they still evaluate as not equal.